app/models/auth_sources/auth_source_ldap.rb

```diff
@@ -130,6 +131,28 @@ def users_in_group(name)
     ldap_con.user_list(name)
   end
 
+  def test_connection
+    result = {}
+    begin
+      Timeout::timeout(10) do
+        self.ldap_con.valid_user?("")
```

Comment:
There's a [test method](https://github.com/theforeman/ldap_fluff/blob/master/lib/ldap_fluff/ldap_fluff.rb#L32) you can use.